### ISSUE TITLE
How to freeze the weights ?

### ISSUE THREAD

**Issue by Da-Bean** [rstudio/keras#717]
Hi guys,

I have a question about freezing the weights.
I have tried to freeze the weights.
But, I can`t freeze the weight.

library(keras)
library(tensorflow)

input_layer<-layer_input(shape = c(943))

encoded<- input_layer %>%
  layer_dense(units=3000) %>%
  layer_activation_leaky_relu(alpha = 0.7)

decoded<- encoded %>%
  layer_dense(units = 943) %>%
  layer_activation_leaky_relu(alpha = 0.7)

Autoencoder<-keras_model(inputs = input_layer,outputs = decoded)

encoder<-keras_model(input_layer,encoded)  

Autoencoder %>% compile(
  loss='mean_squared_error',
  optimizer=optimizer_sgd(lr=0.1,momentum = 0.9,decay = 1e-6),
  metrics='mae'
)

summary(Autoencoder)

Autoencoder %>%fit(
  Geo_Data_Lm_tr,
  Geo_Data_Lm_tr,
  epochs=3,
  shuffle=TRUE,
  validation_data=list(Geo_Data_Lm_va,Geo_Data_Lm_va),
  batch_size=200
)

d<-get_weights(encoder)
freeze_weights(encoder) #1

Regression<-encoder$output %>%
  layer_dense(units=9000,activation='tanh') %>%
  layer_dropout(0.1) %>%
  layer_dense(units=9000,activation='tanh') %>%
  layer_dropout(0.1) %>%
  layer_dense(units=9000,activation='tanh') %>%
  layer_dropout(0.1) %>%
  layer_dense(units=9520)

Regression_model<-keras_model(inputs = encoder$input,outputs = Regression)

Regression_model %>% compile(
  loss='mean_squared_error',
  optimizer=optimizer_sgd(lr=0.4,momentum = 0.9,decay = 1e-6),
  metrics='mae'
)
Regression_model %>% fit(
  Geo_Data_Lm_tr,
  Geo_Data_Ta_tr,
  epochs=3,
  shuffle=TRUE,
  validation_data=list(Geo_Data_Lm_va,Geo_Data_Ta_va),
  batch_size=200
)

f<-get_weights(encoder) #2

identical(d,f)

I want to freeze the weight while line #1 to #2.

But when I check the weight of the encoder it is difference to weight line #2.

So, why are they different? please help me.
